src/lib/use-layout.ts

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import RN from 'react-native';
+
+interface Layout {
+  width: number;
+  height: number;
+}
+
+export default function(): [
+  Layout,
+  (event: RN.LayoutChangeEvent) => void | undefined,
```

Comment:
This 'void | undefined' thing is some black magic stuff. I am not sure if this is a bug in TS compiler or what but it is the only way that I am aware of to ensure that your function returns undefined and only undefined and does not explicitly return anything.

So, if you have a function that is () => 12, you can pass it to a prop that takes type () => void but you cannot pass it to a type () => void | undefined

This is very important when using Reacts useEffect hook. It has bitten me once.